### ISSUE TITLE
Items removed from afArrayField leave null entries in document passed to formToDoc

### ISSUE THREAD

**Issue by mologie** [aldeed/meteor-autoform#816]
Hello, I found that `afArrayField` leaves "null" entries in the document's arrays passed to `formToDoc` after adding, then removing entries, as illustrated by the following example. I did not find a description of this behaviour in the docs and it did just bite me by causing an exception.

```js
var peopleSchema = new SimpleSchema({
    name: { type: String }
});

var businessSchema = new SimpleSchema({
    people: { type: [peopleSchema] }
});

var businesses = new Mongo.Collection("businesses");
businesses.attachSchema(businessSchema);

// ... place a simple quickForm which inserts into "businesses" in the HTML

AutoForm.addHooks(["businessEditSchema"], {
    formToDoc: function (form) {
        console.log(form);
        return form;
    }
});
```

1. Fill in the person's name. `doc.people` is reported as `[{name: "Test"}]`, all good
2. Hit the remove-button. `doc.people` is reported as `[null]`
3. Add the person back. `doc.people` is `[null,{name: "Test"}]`

**Comment by aldeed** [comment]
I think I caused this bug when I moved the compactArrays call [here](https://github.com/aldeed/meteor-autoform/blob/master/autoform-api.js#L312-L335). Will fix

**Comment by garenor** [comment]
Running against the same problem, but I am also ending up with the following:

````
[
    0: "this.0",
    1: null,
    2: {myObject}

]
````
Not only ending up with null, but the `this.0` string should be object 0;

**Comment by aldeed** [comment]
Fixed in 5.1.0. In order to fix this, I had to create a `formToModifier` hook. That will be called instead of `formToDoc` for "update" and "method-update" forms, and it's passed a modifier.
